### PR TITLE
feat: add tidb backend

### PR DESCRIPTION
**Pull request by YikSanChan**

### CODE REVIEW

File: internal/database/metadata/mysql/schema.go [hunk 1]
```diff
@@ -55,20 +55,13 @@ var META_TABLE_SCHEMAS = map[string]string{
...
-	"feature_group": `
-		ALTER TABLE feature_group
-		ADD FOREIGN KEY (entity_id) REFERENCES entity(id),
-		ADD FOREIGN KEY (online_revision_id) REFERENCES feature_group_revision(id)
-	`,
-	"feature_group_revision": `
-		ALTER TABLE feature_group_revision
-		ADD FOREIGN KEY (group_id) REFERENCES feature_group(id)
-	`,
+var META_TABLE_FOREIGN_KEYS = []string{
```
Comment (YikSanChan) [inline internal/database/metadata/mysql/schema.go:58]: it doesn't need to be a `map`. Shall we modify other metadata schemas as well?

Comment (lianxmfor) [reply internal/database/metadata/mysql/schema.go:58]: Using `slice` does accomplish the same thing, but `map` makes the code clearer and more readable.

Comment (YikSanChan) [reply internal/database/metadata/mysql/schema.go:58]: We can't use a map here since there are 2 constraints for a same table

**Review by lianxmfor — APPROVED** [review]
lgtm

File: internal/database/metadata/mysql/schema.go (hunk 1 above)
Comment (lianxmfor) [reply internal/database/metadata/mysql/schema.go:58]: o-o there `slice` is fine.

**Review by wfxr — APPROVED** [review]
LGTM

**Review by jinghancc — APPROVED** [review]
LGTM

**Pull request merged**